src/Interpreters/Cache/FileCache.h

```diff
@@ -102,11 +102,11 @@ class FileCache : private boost::noncopyable
 
     bool tryReserve(FileSegment & file_segment, size_t size);
 
-    FileSegmentsHolderPtr getSnapshot();
+    FileSegments getSnapshot();
 
-    FileSegmentsHolderPtr getSnapshot(const Key & key);
+    FileSegments getSnapshot(const Key & key);
 
-    FileSegmentsHolderPtr dumpQueue();
+    FileSegments dumpQueue();
```

Comment:
Is it okay to make a copy of the entire list? At first glance, seems expensive.

Reply:
`dumpQueue` is used for unit test only, and `getSnapthot` is only for system tables. Also afaik the the result is moved and not copied as the function returns a local variable.